src/main/webapp/app/exercises/quiz/manage/quiz-exercise.component.ts

```diff
@@ -158,14 +158,12 @@ export class QuizExerciseComponent extends ExerciseComponent {
      * @param quizExerciseId the quiz exercise id to end
      */
     endQuiz(quizExerciseId: number) {
-        this.quizExerciseService.end(quizExerciseId).subscribe({
+        return this.quizExerciseService.end(quizExerciseId).subscribe({
             next: (res: HttpResponse<QuizExercise>) => {
                 this.handleNewQuizExercise(res.body!);
+                this.dialogErrorSource.next('');
             },
-            error: (res: HttpErrorResponse) => {
-                this.onError(res);
-                this.loadOne(quizExerciseId);
```

Comment:
The quiz is no longer loaded again in the event of an error. Is this intended? Do you know why it was done before?

Reply:
yes this intended because it breaks the confirm box if the button it belonged to is gone. And if there was an error the exercise will have changed state in a way that the action is no longer valid and that button is no longer shown.